src/gtk/toga_gtk/widgets/numberinput.py

```diff
@@ -16,12 +16,13 @@ def create(self):
         self.native.set_adjustment(self.adjustment)
         self.native.set_numeric(True)
 
-        self.native.connect("value-changed", self.gtk_on_change)
+        self.native.connect("changed", self.gtk_on_change)
 
         self.rehint()
 
     def gtk_on_change(self, widget):
-        self.interface._value = Decimal(self.native.get_value()).quantize(self.interface.step)
+        value = widget.get_text().replace(",", ".") or 0
```

Comment:
I'm worried about this being brittle... how confident are we that we can only get numbers and these specific characters? Does GTK itself do upstream validation of the input?

Reply:
`SpinButton` is set to numeric which forces numbers only.
When manually typing the number, the widget only allows one comma do denote decimal digits.  Inputting letters or any other symbol has no result and does not trigger the `changed` signal.
I am not familiar with the GTK 3 inner workings though.

Reply:
I've just tested this on GTK, and what @cachitas says checks out - the only valid characters are numbers, everything else is ignored. In the AU/US locale, even the `,` is rejected; I'm guessing this only came up for @cachitas because they're in EU, and the comma is the decimal separator.